### PR TITLE
Improve fixed-point documentation

### PR DESCRIPTION
**Pull request by ISSOtm**
This PR may be best reviewed commit by commit.

### CODE REVIEW

File: man/rgbasm.5 [hunk 1]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
 delim off
 .EN
 .Pp
-All of these fixed-point functions can take an optional final argument, which is the precision to use.
+Operators not in the list above, such as
+.Sq +
+and
+.Sq - ,
+work correctly with fixed-point operands as well (but
+.Em both
```
Comment (Rangi42) [inline man/rgbasm.5:384]: The example of `1.0 * 2` shows that they don't always both *need* to be fixed-point.

File: man/rgbasm.5 [hunk 2]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
 .EQ
 delim off
 .EN
 .Pp
-All of these fixed-point functions can take an optional final argument, which is the precision to use.
+Operators not in the list above, such as
+.Sq +
+and
+.Sq - ,
+work correctly with fixed-point operands as well (but
```
Comment (Rangi42) [inline man/rgbasm.5:383]: I'm not sure this precisely explains what's going on. Other operands do "work" with whatever numbers you want, the results will just not be interpretable as either the integers or the fixed-point values you might naively expect. (Also the list above has no "operators", just fixed-point functions.)

How about something like:

> There are no functions for fixed-point addition and subtraction because the '`+`' and '`-`' operators already do that when both operands are fixed-point.

> Note that some operators or functions are meaningful when combining integers and fixed-point values. For example, `2.0 * 3` is equivalent to `MUL(2.0, 3.0)`, and `6.0 / 2` is equivalent to `DIV(6.0, 2.0)`. Be careful and think about what the operations mean when doing this sort of thing.

File: man/rgbasm.5 [hunk 3]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
 returns 1 if the operand was 0, and 0 otherwise.
 Even a non-constant operand with any non-zero bits will return 0.
 .Ss Fixed-point expressions
-Fixed-point numbers are basically normal (32-bit) integers, which count fractions instead of whole numbers.
-They offer better precision than integers but limit the range of values.
-By default, the upper 16 bits are used for the integer part and the lower 16 bits are used for the fraction (65536ths).
-The default number of fractional bits can be changed with the
+Fixed-point numbers are basically normal integers, but conceptually they contain a dot at a fixed location\(emhence the name.
```
Comment (Rangi42) [inline man/rgbasm.5:331]:
```suggestion
Fixed-point numbers are technically just integers, but conceptually they have a decimal point at a fixed location (hence the name).
```

File: man/rgbasm.5 [hunk 4]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
+This gives them an increased precision, at the cost of a smaller range, while remaining far cheaper to manipulate than floating-point numbers.
```
Comment (Rangi42) [inline man/rgbasm.5:332]:
```suggestion
This gives them increased precision at the cost of a smaller range, while remaining far cheaper to manipulate than floating-point numbers (which RGBASM does not support).
```

File: man/rgbasm.5 [hunk 5]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
-Some integer operators like
-.Sq +
-and
-.Sq -
-don't care whether the operands are integers or fixed-point.
-You can easily truncate a fixed-point number into an integer by shifting it right by 16 bits.
-It follows that you can convert an integer to a fixed-point number by shifting it left.
-.Pp
-The following functions are designed to operate with fixed-point numbers:
+followed by the number of fractional bits:
```
Comment (Rangi42) [inline man/rgbasm.5:344]: I think "By default" is very misleading here, when the actual *default* is further below.

Suggestion:
> The default precision of all fixed-point numbers is 16 bits, meaning the lower 16 bits are used for the fractional part; so they count in 65536ths of 1.0.
> This precision can be changed with the Q command-line option, or by OPT Q (see "Changing options while assembling").
> An individual fixed-point literal can have its own precision by appending a `q` followed by the number of fractional bits: for example,

File: man/rgbasm.5 [hunk 6]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
+.Ql 1234.5q8
+is equal to $0004d280
 .EQ
 delim $$
 .EN
+($= 1234.5 * 2 sup 8$).
```
Comment (Rangi42) [inline man/rgbasm.5:350]: Using `sup` looks better in HTML, as a proper superscript, but `man` renders it invisible (at least for me, on Linux; not sure what macOS does but I don't trust it either). I think readers would understand an exponent operator.
```suggestion
($= 1234.5 * 2 \[ha] 8$).
```

File: man/rgbasm.5 [hunk 7]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
+.Ql 1.0 + 2 ;
+and use
+.Ql MUL(1.0, 2.0)
+or
+.Ql 1.0 * 2 ,
+not
+.Ql MUL(1.0, 2) ) .
+.Pp
+All of these fixed-point functions can take an optional final argument, which is the precision to use
+.Em for the operation .
```
Comment (Rangi42) [inline man/rgbasm.5:397]:
```suggestion
for that one operation .
```

File: man/rgbasm.5 [hunk 8]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
 For example,
 .Ql MUL(6.0q8, 7.0q8, 8)
 will evaluate to
 .Ql 42.0q8
 no matter what value is set as the current
 .Cm Q
 option.
+Currently,
```
Comment (Rangi42) [inline man/rgbasm.5:405]: Everything in this doc is "current", and there's been no suggestion that this would change in future.
```suggestion
```

File: man/rgbasm.5 [hunk 9]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
+.Nm
+.Em does not check precisions for consistency ,
+so
+.Ql MUL(4.2q8, 6.9q12, 16)
+will produce a nonsensical result without any warnings.
```
Comment (Rangi42) [inline man/rgbasm.5:410]:
```suggestion
so nonsensical input like
.Ql MUL(4.2q8, 6.9q12, 16)
will produce a nonsensical (but technically correct) result: "garbage in, garbage out".
```

File: man/rgbasm.5 [hunk 10]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
 .Bd -literal -offset indent
-; Generate a table of sine values from sin(0.0) to sin(1.0), with
-; amplitude scaled from [-1.0, 1.0] to [0.0, 128.0]
-DEF turns = 0.0
-REPT 256
-    db MUL(64.0, SIN(turns) + 1.0) >> 16
-    DEF turns += 1.0 / 256
+; Generate a table of sine values from sin(0.0) to sin(1.0) excluded,
+; with amplitude scaled from [-1.0, 1.0] to [0.0, 128.0].
+FOR angle, 0.0, 1.0
```
Comment (Rangi42) [inline man/rgbasm.5:423]:
```suggestion
; Generate a table of 256 sine values from sin(0.0) to sin(1.0) excluded,
; with amplitude scaled from [-1.0, 1.0] to [0.0, 128.0].
FOR angle, 0.0, 1.0, 1.0 / 256
```

File: man/rgbasm.5 [hunk 11]
```diff
@@ -328,76 +328,100 @@ with a non-zero constant as either operand will be constant 1, even if the other
...
-The trigonometry functions (
-.Ic SIN ,
-.Ic COS ,
-.Ic TAN ,
-etc) are defined in terms of a circle divided into 1.0 "turns" (equal to 2pi radians or 360 degrees).
+The trigonometry functions
+.Pq Ic SIN , Ic COS , Ic TAN , No etc
+are defined in terms of a circle divided into 1.0
+.Dq turns
+(equal to 2pi radians or 360 degrees).
```
Comment (Rangi42) [inline man/rgbasm.5:416]:
```suggestion
(equal to $2 pi$ radians or 360 degrees).
```

File: man/rgbasm.5 [hunk 12]
```diff
@@ -355,17 +355,21 @@ The following functions are designed to operate with fixed-point numbers:
 .It Fn DIV x y Ta Fixed-point division
 .It Fn MUL x y Ta Fixed-point multiplication
 .It Fn FMOD x y Ta Fixed-point modulo
-.It Fn POW x y Ta $x$ to the $y$ power
-.It Fn LOG x y Ta Logarithm of $x$ to the base $y$
+.It Fn POW x y Ta $x sup y$
```
Comment (Rangi42) [inline man/rgbasm.5:358]: Double-check how this renders in `man` pages.

**Review by Rangi42 — CHANGES_REQUESTED** [review]

File: man/rgbasm.5 (hunk 1 above)
Comment (ISSOtm) [reply man/rgbasm.5:384]: That's the hard part to explain: `*` works fine iff *either operand* is *not* fixed-point, whereas `+` works iff *neither* operand is fixed-point. It's a bit of a headache (but that's mathematical properties for you lololol), and thus AFAICT a headache to explain as well.

I wonder if @aaaaaa123456789 has any ideas about this.

File: man/rgbasm.5 (hunk 2 above)
Comment (ISSOtm) [reply man/rgbasm.5:383]: I wholeheartedly agree with the second block, but for the first one, I wonder if there are any operators we're missing?

(Additionally, shouldn't `%` work just as well as `FMOD`?)

File: man/rgbasm.5 (hunk 4 above)
Comment (ISSOtm) [reply man/rgbasm.5:332]: `which neither .Nm nor the Game Boy support`, maybe? (I want to point out the lack of a FPU.)

File: man/rgbasm.5 (hunk 5 above)
Comment (ISSOtm) [reply man/rgbasm.5:344]: (To be fair, this was also a problem with the original text :P)

File: man/rgbasm.5 (hunk 1 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:384]: At its heart, the issue here is how factors combine: this is a result of ab + ac = a(b + c) and ab \* ac = a²(bc). I wonder how much you want to turn this page into a tutorial on fixed point: I fear that an overload of examples would have the wrong effect, teaching people to replicate expressions like cargo cult without understanding _why_ the expressions are behaving like they do. After all, the examples are endless: `1.5q8 * 2.5q8` is `3.75q16`.

File: man/rgbasm.5 (hunk 6 above)
Comment (ISSOtm) [reply man/rgbasm.5:350]: Interesting, that would be a bug in `groff`, then. Mandoc renders this as `= 1234.5 * 2^8`.

(Note that I specifically avoided `<<`, because most calculator programs tend to truncate the LHS to an integer before performing the operation.)

File: man/rgbasm.5 (hunk 8 above)
Comment (ISSOtm) [reply man/rgbasm.5:405]: I think it *would* be worth warning about.

File: man/rgbasm.5 (hunk 4 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:332]: No, do not. By your idea of "does not support", the Game Boy doesn't even support multiplication!

File: man/rgbasm.5 (hunk 10 above)
Comment (ISSOtm) [reply man/rgbasm.5:423]: Oops, I actually had the iteration step in my local testing copy, but forgot it when transcribing :P

File: man/rgbasm.5 (hunk 1 above)
Comment (Rangi42) [reply man/rgbasm.5:384]: I just don't think "works fine" is a precise enough explanation. `5.0 * 3 == 5 * 3.0 == 15.0`; that's not some arbitrary property we gave the `*` operator, it's a consequence of how multiplication acts. `5.0 + 1` "works" if you wanted the 1 to be short for 1/Q.

Frankly I'm not sure we even need to point out how `*` acts here; we already provide `MUL`, and we explain that `fixed + fixed` is how you add fixed-point. Advanced users who *want* to combine fixed with integer will soon figure out how that behaves.

File: man/rgbasm.5 (hunk 2 above)
Comment (Rangi42) [reply man/rgbasm.5:383]: I don't see anything in the [operators table](https://rgbds.gbdev.io/docs/master/rgbasm.5#Operators) that ought to have a fixed-point function.

I'm not sure if `FMOD` behaves entirely the same as `%`. Maybe? Anyway, we already provide the function.

File: man/rgbasm.5 (hunk 4 above)
Comment (Rangi42) [reply man/rgbasm.5:332]: I think the Pan Docs are the place for discussing the Game Boy's (lack of) FPU. :P

File: man/rgbasm.5 (hunk 2 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:383]: %

Comment (aaaaaa123456789) [reply man/rgbasm.5:383]: `FMOD` and `%` should be functionally identical. Quick test: `FMOD(x, 1.0)` will keep the fractional bits, and so will `x % 1.0`.

File: man/rgbasm.5 (hunk 8 above)
Comment (Rangi42) [reply man/rgbasm.5:405]: Then that can get its own issue, discussion, and maybe a PR in which this doc would be updated.

File: man/rgbasm.5 (hunk 2 above)
Comment (Rangi42) [reply man/rgbasm.5:383]: Even with non-integral or negative operands, on either side?

Comment (aaaaaa123456789) [reply man/rgbasm.5:383]: I don't even know what `%` does with negatives. Is that documented? Does it just follow C here?
Non-integral operands should work just fine.

File: man/rgbasm.5 (hunk 12 above)
Comment (ISSOtm) [reply man/rgbasm.5:358]: image

File: man/rgbasm.5 (hunk 4 above)
Comment (ISSOtm) [reply man/rgbasm.5:332]: I fear that people are then going to ask why RGBASM doesn't support floating-point arith, instead of this weird nonsense of half-different operators.

Comment (aaaaaa123456789) [reply man/rgbasm.5:332]: Which is a legitimate question! The platform where RGBASM runs certainly supports floats, so RGBASM easily could. This is a design decision, not a limitation of any kind, and there's no reason to lie to people about it.

File: man/rgbasm.5 (hunk 12 above)
Comment (Rangi42) [reply man/rgbasm.5:358]: Yeah, <code><em>xy</em></code> looks like multiplication, not exponentiation. :P

File: man/rgbasm.5 (hunk 6 above)
Comment (ISSOtm) [reply man/rgbasm.5:350]: Considering that the documentation tends to be primarily read in HTML (online), that `groff` may eventually fix their rendering, and that the meaning of `POW` is relatively obvious anyway, I'd prefer to keep this as-is.

File: man/rgbasm.5 (hunk 1 above)
Comment (ISSOtm) [reply man/rgbasm.5:384]: Then perhaps “give the expected results if used with two fixed-point oeprands”?

File: man/rgbasm.5 (hunk 4 above)
Comment (Rangi42) [reply man/rgbasm.5:332]: #956 was a serious feature request for floating-point support in RGBASM. And fixed-point support isn't something new; it's been an RGBASM feature for years. It's nice to be cleaning up the docs for it, but it's not like we're preparing a surprising and confusing new feature for people.

File: man/rgbasm.5 (hunk 2 above)
Comment (Rangi42) [reply man/rgbasm.5:383]: Yes, we document it to do the opposite of C:

> `%` is used to get the remainder of the corresponding division, so that `a / b * b + a % b == a` is always true. The result has the same sign as the divisor. This makes `a % b` equal to `(a + b) % b` or `(a - b) % b`.

`FMOD()` works like IEEE-754 `fmod()`.

File: man/rgbasm.5 (hunk 6 above)
Comment (Rangi42) [reply man/rgbasm.5:350]: Ehhh. We've generally been careful about writing these man-page macros to render well *as* man pages. (If we're not expecting users to actually notice/care, maybe we can finally switch to HTML or Markdown for them?)

File: man/rgbasm.5 (hunk 2 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:383]: In that case it's probably slightly different for negatives. I don't remember which one of the two common definitions IEEE uses, but I'm pretty sure it's not that one.

File: man/rgbasm.5 (hunk 8 above)
Comment (ISSOtm) [reply man/rgbasm.5:405]: Agreed; but, in the meantime, should this be kept?

Comment (Rangi42) [reply man/rgbasm.5:403]: (We don't say "Currently these are all the functions that exist[, but we have ideas to Eventually let users define their own]"; or "Currently a charmap is a single byte[ but we've had a feature request for years to extend that]".)

File: man/rgbasm.5 (hunk 1 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:382]: A lot of this discussion is falling into "I wish this was a different project" territory. (It comes up often: for instance, some people wish there was a GB standard library, but there isn't, so they attempt to stuff unneeded stuff into hardware.inc instead...)
Namely, a lot of this is attempting to be a fixed-point tutorial, which the docs should never even attempt. That would make them terrible (due to size alone, if nothing else) as documentation for users who just want to learn the RGBASM behaviors.

File: man/rgbasm.5 (hunk 6 above)
Comment (ISSOtm) [reply man/rgbasm.5:348]: I agree, but degrading our primary UX for the sake of working around a *bug* in `groff` also makes me go “Ehhh.”

File: man/rgbasm.5 (hunk 12 above)
Comment (ISSOtm) [reply man/rgbasm.5:356]: [Discussion of exponent rendering in an earlier thread](https://github.com/gbdev/rgbds/pull/1455/files#r1706839210)

File: man/rgbasm.5 (hunk 1 above)
Comment (ISSOtm) [reply man/rgbasm.5:382]: Sure, but the goal here is to explain *why* there's no `ADD()` or `SUB()`. (We could add them, honestly.)

File: man/rgbasm.5 (hunk 6 above)
Comment (Rangi42) [reply man/rgbasm.5:350]: On the other hand, if `man 5 rgbasm` is meant to be legible, then seeing `xy` when you want `x^y` is a lot more of a degradation (a step backwards from how it used to be!) than keeping <em>x</em>^<em>y</em> instead of <em>x<sup>y</sup></em>.

File: man/rgbasm.5 (hunk 1 above)
Comment (Rangi42) [reply man/rgbasm.5:384]: See the comment below:
> There are no functions for fixed-point addition and subtraction because the '+' and '-' operators already do that when both operands are fixed-point.
Easy-peasy.
You could reword it to:
> There are no functions for fixed-point addition and subtraction,  because the '+' and '-' operators can add and subtract pairs of fixed-point operands.
Or something like that.

File: man/rgbasm.5 (hunk 6 above)
Comment (aaaaaa123456789) [reply man/rgbasm.5:350]: I'm personally much more likely to open a man page than a browser...

File: man/rgbasm.5 (hunk 1 above)
Comment (Rangi42) [reply man/rgbasm.5:384]: No please let's *not* add redundant functions just so everything fixed-point can look function-y.

It wouldn't simplify documentation -- it would make the bikeshedding worse, as "ADD(x, y) Fixed-point addition" would be a falsehood, and "ADD(x, y) Fixed-point addition (which is the same as integer addition, but this function is included for ??? purposes)" would be a weirdly bloated documentation of an exceptionally simple fixpoint operation.

(Also `add` and `sub` are opcodes, are we certain there'd be no parsing conflicts?)

Comment (ISSOtm) [reply man/rgbasm.5:384]: > You could reword it to:
> 
> > There are no functions for fixed-point addition and subtraction,  because the '+' and '-' operators can add and subtract pairs of fixed-point operands.

Looks good to me!

---

<details><summary>Results of the nerd-snipe:</summary>

(There would be no parsing conflicts, because instructions and expressions are never expected in the same context. *One of the few nice things about a language that doesn't have expression-as-statement?* :stuck_out_tongue:)

</details>

File: man/rgbasm.5 (hunk 2 above)
Comment (Rangi42) [reply man/rgbasm.5:383]: Sounds good to me! (It had better, since I wrote those comments. :P )

File: man/rgbasm.5 (hunk 8 above)
Comment (pinobatch) [reply man/rgbasm.5]: I'd prefer to drop *all* "currently" and the like, given [Google's guidance about timeless documentation](https://developers.google.com/style/timeless-documentation). When comparing of features between supported versions, use language like the notes in the Python library reference that begin with "Added in version 3.2" or "Changed in version 3.7".

Comment (aaaaaa123456789) [reply man/rgbasm.5]: "Do it because Google does it" is the single worst argument I've seen in this thread, for any side :P

Comment (Rangi42) [reply man/rgbasm.5]: As far as I know, we have never yet *had* any time-anchored language; "current" appears a lot but in regard to the state of RGBASM/LINK during processing.

File: man/rgbasm.5 (hunk 6 above)
Comment (pinobatch) [reply man/rgbasm.5:350]: One practical problem with an HTML manual is that when a program is installed for one user, its HTML manual ends up inside `$HOME/.local/share`. Like other hidden folders, `$HOME/.local` is off-limits to web browsers in a Snap package. The default web browser under major Ubuntu flavors is Firefox in a Snap package. [This limit of Snap has already caused people to be unable to view Rust documentation](https://askubuntu.com/questions/1419528/how-do-i-access-rust-documentation).

Another is that a lot of these static site generators used for building an HTML manual from source code rely on Node.js. This would significantly increase the total dependency download size and total dependency install footprint of RGBDS on all GNU/Linux distributions that . We would need to carefully consider make Node.js and NPM into dependencies for building and installing RGBDS.

File: man/rgbasm.5 (hunk 8 above)
Comment (ISSOtm) [reply man/rgbasm.5]: Then maybe it's a good idea not to start.

Comment (Rangi42) [reply man/rgbasm.5]: Whoops, there is *one*:
> The options that OPT can modify are currently: b, g, p, Q, and r.

Comment (pinobatch) [reply man/rgbasm.5]: I meant not "because consistency with Google style is desirable" but "for the reasons described in this article, which happens to be published by Google."

File: man/rgbasm.5 (hunk 2 above)
Comment (ISSOtm) [reply man/rgbasm.5:383]: Applied that change.

File: man/rgbasm.5 (hunk 6 above)
Comment (ISSOtm) [reply man/rgbasm.5:350]: Ongoing discussion in #1457; discussion there agrees that the non-CLI docs are primarily viewed in the browser, so I'm going to privilege the change that improves *that*. (And try reporting the bug to `groff`.)

**Comment by ISSOtm** [comment]
(Rebased on top of #1450.)

File: man/rgbasm.5 [hunk 13]
```diff
@@ -369,63 +369,93 @@ equals $roman clz ( n )$.
...
+command-line option, and/or by
+.Ic OPT Q
+.Pq see Sx Changing options while assembling .
+An individual fixed-point literal can have its own precision overridden by appending a
 .Dq q
-to it followed by the number of fractional bits, such as
-.Ql 12.34q8 .
+followed by the number of fractional bits: for example,
+.Ql 1234.5q8
+is equal to $0004d280
```
Comment (Rangi42) [inline man/rgbasm.5:387]: May as well set a good example of how to use `_`.
```suggestion
is equal to $4d2_80
```

File: man/rgbasm.5 [hunk 14]
```diff
@@ -369,63 +369,93 @@ equals $roman clz ( n )$.
...
+.Pp
+The default precision of all fixed-point numbers is 16 bits, meaning the lower 16 bits are used for the fractional part; so they count in 65536ths of 1.0.
+This precision can be changed with the
 .Fl Q
-command-line option.
-You can also specify a precise fixed-point value by appending a
+command-line option, and/or by
+.Ic OPT Q
+.Pq see Sx Changing options while assembling .
+An individual fixed-point literal can have its own precision overridden by appending a
```
Comment (Rangi42) [inline man/rgbasm.5:383]:
```suggestion
An individual fixed-point literal can specify its own precision, overriding the current default, by appending a
```

File: man/rgbasm.5 [hunk 15]
```diff
@@ -369,63 +369,93 @@ equals $roman clz ( n )$.
...
-.It Fn MUL x y Ta Fixed-point multiplication $( x \[mu] y ) \[di] ( 2 ^ precision )$
-.It Fn FMOD x y Ta Fixed-point modulo $( x % y ) \[di] ( 2 ^ precision )$
-.It Fn POW x y Ta $x$ to the $y$ power
-.It Fn LOG x y Ta Logarithm of $x$ to the base $y$
+.It Fn DIV x y Ta Fixed-point division
+.It Fn MUL x y Ta Fixed-point multiplication
+.It Fn FMOD x y Ta Fixed-point modulo
+.It Fn POW x y Ta $x sup y$
+.It Fn LOG x y Ta $log sub y ( x )$ (such that
+.Ic LOG Ns ( Ic POW Ns ( x , y ) , y ) == x)
```
Comment (Rangi42) [inline man/rgbasm.5:408]: If we're removing the low-level explanation of how fixed-point `MUL` and `DIV` work, why add one for `LOG`? Anyone who needs one will know what "log to a base" is; it's a mathematical fact, not anything specific to fixed-point.
```suggestion
.It Fn LOG x y Ta Logarithm of $x$ to the base $y$
```
(Likewise for the inverse trig functions, but no need to split discussion among comments.)

File: man/rgbasm.5 [hunk 16]
```diff
@@ -369,63 +369,93 @@ equals $roman clz ( n )$.
...
 .El
 .EQ
 delim off
 .EN
 .Pp
-All of these fixed-point functions can take an optional final argument, which is the precision to use.
+There are no functions for fixed-point addition and subtraction, because the
+.Sq +
+and
+.Sq - ,
```
Comment (Rangi42) [inline man/rgbasm.5:430]:
```suggestion
.Sq -
```

**Review by Rangi42 — DISMISSED** [review]

File: man/rgbasm.5 (hunk 13 above)
Comment (ISSOtm) [reply man/rgbasm.5:387]: I padded it to highlight that the entire number is still 32-bit. What do you prefer between this and
```suggestion
is equal to $0004d2_80
```
?

Comment (Rangi42) [reply man/rgbasm.5:387]: `$0004d2_80` looks good!

File: man/rgbasm.5 (hunk 15 above)
Comment (ISSOtm) [reply man/rgbasm.5:408]: Because those are not low-level explanations, but mentions of properties that they hold.

Comment (aaaaaa123456789) [reply man/rgbasm.5:408]: It still falls under the "explaining math" umbrella — this is the definition of a logarithm, not something specific to RGBASM.

Comment (ISSOtm) [reply man/rgbasm.5:408]: In theory yes, but I've seen different notation used for at least the trig functions (<code>sin<sup>-1</sup></code>, `isin` for “inverse”...), so I think it's warranted there.

`log` should be universal enough that I agree with removing this explanation.

File: man/rgbasm.5 [hunk 17]
```diff
@@ -369,63 +369,93 @@ equals $roman clz ( n )$.
...
 .It Fn TAN x Ta Tangent of $x$
-.It Fn ASIN x Ta Inverse sine of $x$
-.It Fn ACOS x Ta Inverse cosine of $x$
-.It Fn ATAN x Ta Inverse tangent of $x$
+.It Fn ASIN x Ta Inverse sine of $x$ (such that
+.Ic ASIN Ns ( Ic SIN Ns ( x ) ) == x)
+.It Fn ACOS x Ta Inverse cosine of $x$ (such that
+.Ic ACOS Ns ( Ic COS Ns ( x ) ) == x)
+.It Fn ATAN x Ta Inverse tangent of $x$ (such that
+.Ic ATAN Ns ( Ic TAN Ns ( x ) ) == x)
```
Comment (Rangi42) [inline man/rgbasm.5:420]: From the `LOG` discussion:

> I've seen different notation used for at least the trig functions (`sin-1`, `isin` for “inverse”...), so I think it's warranted there.

There's no ambiguity about the explanation "inverse sine", and just calling it `ASIN` instead of `SINM1` or `ISIN` shows which convention we're using.

Comment (ISSOtm) [reply man/rgbasm.5:420]: Having grown up in a country where I wasn't even *aware* that the `arc*` notation *existed*, I'm feeling iffy about this.

The equations help disambiguate that we're talking about “the reverse of `func`” and not “`1 / func`”, since the latter is also called an inverse.

Comment (aaaaaa123456789) [reply man/rgbasm.5:420]: I recall a similar incident where you thought that 4 was smaller than 4... I do wonder if your (as in, your area's) math terminology is just uncommon.
Arc sine and friends are the standard English names for these functions — if someone doesn't know them, a web search would answer the question instantly.

Comment (ISSOtm) [reply man/rgbasm.5:420]:
```suggestion
.It Fn ASIN x Ta Inverse sine of $x$
.It Fn ACOS x Ta Inverse cosine of $x$
.It Fn ATAN x Ta Inverse tangent of $x$
```

Comment (Rangi42) [reply man/rgbasm.5]: How about "arc sinus"? :P <https://fr.wiktionary.org/wiki/arc_sinus>

**Review by Rangi42 — APPROVED** [review]
Thank you for persevering through all the feedback!
